dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/advanced/DateRangeValidator.java

```diff
@@ -50,13 +51,11 @@ public void validate(FacesContext context, UIComponent component, Object value)
             return Optional.empty();
         }
 
-        try {
-            isDatePatternCorrect(value);
-            isDateValueCorrect(value);
-        } catch(ParseException  pe) {
+        if(isDatePatternCorrect(value).isLeft() || isDateValueCorrect(value).isLeft()) {
```

Comment:
That's better than previous versions I think.
Maybe it would be possible to transform the rest of method in more functional way with either, but as for now it's ok.